src/FileWriter.php

```diff
@@ -25,13 +29,18 @@ final class FileWriter
     public static function writeFile(string $file, string $content, int $chmod = 0666) : void
     {
         $dir = dirname($file);
+
         // suppress notice thrown when falling back to system temp dir
-        $tmp = @tempnam($dir, 'wsw');
+        $tmp = self::suppress(static function () use ($dir) {
+            return tempnam($dir, 'wsw');
+        });
 
         if ($tmp === false) {
             throw Exception\RuntimeException::unableToCreateTemporaryFile($dir);
         }
 
+        assert(is_string($tmp));
```

Comment:
If the above block is simplified to `if (! \is_string($tmp)) {` then this assertion can be removed